src/connection/host_lookup.c

```diff
@@ -0,0 +1,622 @@
+    {
+      ret_val = EINVAL;
+
+      goto return_phase;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+  *result = hp_buf;
+
+  ret_val = 0;
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      ret_val = NULL;
+
+      goto return_phase;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  ret_val = ret;
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      ret_val = -1;
+
+      goto return_phase;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  ret_val = 0;
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif /* HAVE_GETHOSTBYNAME_R_GLIBC */
+#endif /* HAVE_GETHOSTBYNAME_R */
+
+return_phase:
+  return ret_val;
+}
+
+
+int
+getnameinfo_uhost (struct sockaddr *addr, socklen_t addrlen, char *host, size_t hostlen, char *serv, size_t servlen,
+		   int flags)
+{
+  int ret;
+  struct hostent *hp = NULL;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      ret = getnameinfo (addr, addrlen, host, hostlen, serv, servlen, flags);
+    }
+  else
+    {
+      if ((hp = host_lookup_internal (NULL, addr, IPADDR_TO_HOSTNAME)) != NULL)
+	{
+	  strncpy (host, hp->h_name, hostlen);
+	  host[hostlen] = '\0';
+
+	  ret = 0;
+	}
+      else
+	{
+	  ret = EAI_NONAME;
+	}
+    }
+  return ret;
+}
+
+int
+getaddrinfo_uhost (char *node, char *service, struct addrinfo *hints, struct addrinfo **res)
+{
+  int ret = 0;
+  struct hostent *hp = NULL;
+  struct addrinfo *addrp;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      ret = getaddrinfo (node, service, hints, res);
+      goto return_phase;
+    }
+  else
+    {
+      hp = host_lookup_internal (node, NULL, HOSTNAME_TO_IPADDR);
+    }
+
+  if (hp == NULL)
+    {
+      ret = EAI_NONAME;
+      goto return_phase;
+    }
+
+  if ((addrp = (struct addrinfo *) malloc (sizeof (struct addrinfo))) == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (struct addrinfo));
+      ret = EAI_MEMORY;
+      goto return_phase;
+    }
+
+  memset (addrp, 0, sizeof (addrinfo));
+  if ((addrp->ai_canonname = strdup (hp->h_name)) == NULL)
+    {
+      freeaddrinfo (addrp);
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (struct sockaddr));
+      ret = EAI_MEMORY;
+      goto return_phase;
+    }
+
+  if (hints != NULL)
+    {
+      addrp->ai_flags = hints->ai_flags;
+      addrp->ai_family = hints->ai_family;
+      addrp->ai_socktype = hints->ai_socktype;
+      addrp->ai_protocol = IPPROTO_TCP;
+    }
+  else
+    {
+      addrp->ai_flags = (AI_V4MAPPED | AI_ADDRCONFIG);
+      addrp->ai_family = AF_UNSPEC;
+    }
```

Comment:
```suggestion
addrp->ai_family = hints ? hints->ai_family : (AI_V4MAPPED | AI_ADDRCONFIG);
```

Reply:
```suggestion
  if (hints != NULL)
    {
      addrp->ai_family = hints->ai_family;
    }
  else
    {
      addrp->ai_family = AF_UNSPEC;
    }
```
@kisoo-han 
Can I use this structure?
It is not related preference with of x = a ? b : c.

1. This parts can be extended. Someday a developer can use  getaddrinfo again and then he or she must fix the  getaddrinfo too. 
At that time, 'if' statement will help that developer.
2. (AI_V4MAPPED | AI_ADDRCONFIG) is flags. So I will change it as AF_UNSPEC.